### ISSUE TITLE
Iterable yield types do not produce errors in generators

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#475]
**Describe the bug**
Iterables are implicitly cast to the yield type in generator functions.  The following:

```py
# pyright: strict

from typing import Iterable


def foo() -> Iterable[str]:
    yield {1, 2}
```

does not produce an error though it clearly should.

**To Reproduce**
Type check the snippet above.

**Expected behavior**
That I should not be able to assign a set of integers to a yield type of string.

**Screenshots or Code**
See above.

**VS Code extension or command-line**
extension, 1.1.17

**Comment by erictraut** [comment]
This should be addressed in 1.1.18 already. I confirmed that it now flags the above as an error. If you are still seeing issues, please reopen.
